app/views/unattended/provisioning_templates/PXEGrub/pxegrub_global_default.erb

```diff
@@ -5,14 +5,19 @@ model: ProvisioningTemplate
 %>
 <%# Used to boot unknown hosts, do not associate or change the name. %>
 <%
-  # Grub1 only supports numeric default statement, this array allows conversion to number
-  items = ["local", "discovery"]
-  default_item = items.index(global_setting("default_pxe_item_global", "local")) || 0
+  # Grub1 only supports numeric default statement, this allows conversion to number. First define
+  # array of directories we will search for EFI booloaders:
+  paths = ["fedora", "redhat", "centos", "debian", "ubuntu", "sles", "opensuse", "Microsoft", "EFI"]
+  # Add remaining entries to it and use this to convert to number:
+  items = paths.push("discovery")
+  # Read default setting but since "local" is missing, use the first path available.
+  default_setting = global_setting("default_pxe_item_global", paths.first).sub("local", paths.first)
```

Comment:
I know I'm a bit difficult here, but the sub read bad + would act weird if the custom setting is something like `my_local_boot`. Anyway, as we default to 0 if the `default_setting` is not in `items`, could we just leave it out without the sub. Otherwise, I would recommend doing something like:

```ruby
default_setting = paths.first if default_setting == 'local'
```

Reply:
Sure sub is unclear.